Automattic/simplenote-electron
# Updates collaborator email field validation

### Fix

Fixes #234 Fixes #233 

The email field for collaborators allowed for incomplete emails. When entering
an incomplete email it would add that partial as a tag on the note. This fix
also disallows the user from adding themselves as a collaborator.

### Test
1. Go to collaborators dialog
2. Add `foo@` see that it is not allowed
3. Add your email, see that it is not allowed
4. Goto a not that is shared with you, add your email, see that it is not allowed

This PR does not cover the case where a not is shared with someone and that user adds the owner as a collaborator. That will require more work as part of: https://github.com/Automattic/simplenote-electron/issues/1728

### Review
Only one developer ais required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated with:

- Disallow partial emails and users own email from being adding to collaborators email field [#1735](https://github.com/Automattic/simplenote-electron/pull/1735)
